client/src/modules/vouchers/modals/search.modal.js

```diff
@@ -26,22 +26,25 @@ bhConstants, Periods, Store, util, TransactionTypes, $translate) {
     'reference', 'description', 'user_id', 'type_id',
   ];
 
-  TransactionTypes.read()
-      .then(function (transactionTypes) {
-        transactionTypes.forEach(function(item){
-          item.plainText = $translate.instant(item.text);
-        });
-        vm.transactionTypes = transactionTypes;
-      })
-      .catch(Notify.handleError);
-
   // assign already defined custom filters to searchQueries object
   vm.searchQueries = util.maskObjectFromKeys(filters, searchQueryOptions);
+  
+  if(vm.searchQueries.type_id){
+    vm.type_ids = [vm.searchQueries.type_id];
+  }
 
   if(filters.limit){
     vm.defaultQueries.limit = filters.limit;
   }
 
+  vm.onSelectTransactionType = function onSelectTransactionType (transactionTypes){
+    vm.searchQueries.type_id = transactionTypes[transactionTypes.length - 1];
```

Comment:
Could you just update the server so that this component works correctly with the server?